### ISSUE TITLE
Error launching source instance: timeout while waiting for state to become 'success' (timeout: 15s)

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#316]
_This issue was originally opened by @CloudSurgeon as hashicorp/terraform#9450. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

### Terraform Version

Terraform v0.7.6 & v0.7.7
### Affected Resource(s)
- aws_instance
### Terraform Configuration Files

https://www.dropbox.com/s/p7zre062k4bnaiu/main.tf?dl=0
### Trace Output

https://gist.github.com/CloudSurgeon/075ef1a8163c126f9c0bd04e3cf631a7
### Expected Behavior

AWS instance creation should not time out when in 'pending' status, or a tunable should exist to adjust this timeout value.
### Actual Behavior

Terrafrom quits with "Error launching source instance: timeout while waiting for state to become 'success' (timeout: 15s)" when the instance takes longer than 15s to go from 'pending' to 'running'
### Steps to Reproduce

Please list the steps required to reproduce the issue, for example:
provision 30 - 40 aws_instances at once. Chances are that one or two will get the aforementioned error.

**Comment by sandyfox** [comment]
Any update on this ?

**Comment by radeksimko** [comment]
Hi @CloudSurgeon & @sandyfox 
the default timeout for launching instance (meaning the time it takes for the instance to reach `running` state) was set to 10mins for a long while (~ 3 years ago) which should be sufficient for vast majority of common cases (instance types & no of instances).

Customizable timeouts were introduced for any corner cases (bigger instance types and/or higher no of instances) in hashicorp/terraform#14711 which was released in [0.9.6 (25th May)](https://github.com/hashicorp/terraform/blob/master/CHANGELOG.md#096-may-25-2017).

Being confused by the low timeout in the attached error message I took time to read the source code to discover we have another timeout in retryer which aims to address eventual consistency nature of the AWS API. That timeout used to be 15 secs and was recently doubled to 30secs in https://github.com/hashicorp/terraform/commit/29444aee87702cbbbfd9d1afa00bb3c50db66b55 (part of 0.9.7 release).

I took your snippet of code and tried spinning up the same number of instances as you did with the latest Terraform build (from master):
`how_many = 9`, so 27 of `m4.large`s + 9 `t2.small`s - that finished successfully.

Do you mind checking if you can reproduce the issue in `0.9.7+` yourself? I was unable to do so, hence I'm inclined to close this.

**Comment by rukumar333** [comment]
I am using 0.9.11 and still running into this issue.

**Comment by radeksimko** [comment]
@rukumar333 Do you mind attaching repro case in the form of configuration files (+ relevant variable values) + console output (with the error message)?

Thanks.

**Comment by rukumar333** [comment]
The log I pasted below. For the configuration files, there are multiple modules set up that reference each other. Which file would you need to see in particular?
>  [2017/08/02 16:15:34.370] Error applying plan:
 [2017/08/02 16:15:34.379] 3 error(s) occurred:
 [2017/08/02 16:15:34.379] * module.cluster.module.configsvr_instance.aws_instance.member[0]: 1 error(s) occurred:
 [2017/08/02 16:15:34.379] * aws_instance.member.0: Error launching source instance: timeout while waiting for state to become 'success' (timeout: 30s)
 [2017/08/02 16:15:34.379] * module.cluster.module.configsvr_instance.aws_instance.member[1]: 1 error(s) occurred:
 [2017/08/02 16:15:34.379] * aws_instance.member.1: Error launching source instance: timeout while waiting for state to become 'success' (timeout: 30s)
 [2017/08/02 16:15:34.379] * module.cluster.module.configsvr_instance.aws_instance.member[2]: 1 error(s) occurred:
 [2017/08/02 16:15:34.379] * aws_instance.member.2: Error launching source instance: timeout while waiting for state to become 'success' (timeout: 30s)

**Comment by andrew-smith** [comment]
For our particular instance - we received this error message when AWS was out of capacity. See comment: https://github.com/hashicorp/terraform/issues/9450#issuecomment-293763876

**Comment by radeksimko** [comment]
@andrew-smith that's a good point. Knowing full error message would help there.

@rukumar333 I'm glad you're still able to reproduce this. Such errors tend to be intermittent 😃 
Regarding configs - I'm specifically interested in any `aws_instance` resource definitions and any significant references, like `count` (i.e. if it comes from a variable then the real value) and `iam_instance_profile` (i.e. `aws_iam_role` + `aws_iam_role_policy` etc.).
Knowing region would also help as some regions can be busier than others (hence more affected).

Do you mind also attaching [debug log](https://github.com/hashicorp/terraform/blob/master/website/source/docs/internals/debugging.html.md#debugging-terraform) (minus any secrets)? I'm specifically interested in any blocks that begin with `DEBUG: Request ec2/RunInstances Details` and `DEBUG: Response ec2/RunInstances Details`.

**Comment by nikuljain** [comment]
Error launching source instance: timeout while waiting for state to become 'success' (timeout: 30s)  - 

trying to launch four instances in Mumbai region

Instance type : t2.large

**Comment by fletchowns** [comment]
I keep running into this with terraform 0.9.11, trying to spin up c3.8xlarge instances in us-west-2. Is there any workaround?

**Comment by radeksimko** [comment]
I'm going to close this issue as I'm unable to reproduce it and we don't have enough data (as asked above) to understand the reasons behind those failures, hence it's impossible to reliably fix it. Feel free to open a new issue if you have some data to share and we can take a look.

Thanks.

**Comment by fededamian** [comment]
It is possible to identify the underlying issue using the **Terraform Debug Logs**. When I had this error, in my case the issue was that AWS did not have enough capacity for the instance type I was choosing in the availability zone I configured.

A good approach to debug this error would be the following:

1. Set up the `TF_LOG` environment variable to **enable Terraform Debug Logs**. This can be done with the following command on Linux or Mac:

```
export TF_LOG=DEBUG
```

2. Run the `terraform apply` command.
3. When the prompt to apply changes appears write `yes` and hit `Enter` if the plan is okay for you.
4. If you get the error, search for the `[DEBUG]` entries a few lines above it to  identify the underlying issue. To give an example, in my case I found the following:

```
[DEBUG] plugin.terraform-provider-aws_v1.58.0_x4:
<Response>
  <Errors>
    <Error>
      <Code>InsufficientInstanceCapacity</Code>
      <Message>We currently do not have sufficient t3.medium capacity in the Availability Zone you requested (us-west-1a). Our system will be working on provisioning additional capacity. You can currently get t3.medium capacity by not specifying an
        Availability Zone in your request or choosing us-west-1b.</Message>
    </Error>
  </Errors>
  <RequestID>...</RequestID>
</Response>
```


5. This should allow you to perform corrective actions in your infrastructure to deploy.
